### PR TITLE
[1.6]Enhance Windows Environment Experience

### PR DESCRIPTION
**Pull request by orangedeng**
refer to https://github.com/rancher/rancher/issues/10442
and https://github.com/rancher/rancher/issues/10448

1. Change the add host step for windows environment. Add setting
subnet labels logic.
2. Run command in Windows environment is fetched from rancher
server.
3. Block the webhook link on windows. Because the webhook is not
fully supported in Windows environment.
4. Because the metadata and DNS services are supported, service
alias and external service can be use in Windows environment.

### CODE REVIEW

**Comment by orangedeng** [comment]
@yasker

File: app/components/machine/driver-custom/component.js [hunk 1]
```diff
@@ -79,11 +110,21 @@ export default Ember.Component.extend(ManageLabels, {
...
-
-    return `New-Item -Path 'C:\\Program Files\\rancher' -Type Directory
-Invoke-WebRequest -UseBasicParsing 'https://github.com/rancher/agent/releases/download/v0.6.0/agent.exe' -OutFile 'C:\\Program Files\\rancher\\agent.exe'
-& 'C:\\Program Files\\rancher\\agent.exe' -register-service ${url}
-Start-Service rancher-agent`;
-  }.property('model.command'),
+    let cattleIp = this.get('cattleAgentIp');
+    let env = Util.addQueryParams('', this.get('model.labels')||{});
+    let cmd1=`New-Item -Path 'C:\\Program Files\\rancher' -Type Directory`;
+    let cmd2 = this.get('model.windows-command');
```
Comment (vincent99) [inline app/components/machine/driver-custom/component.js]: Please tell me this is a typo and there's not really a field called `windows-command` in the API?  JSON fields are camelCase, hyphen is ambiguous and requires quoting in people's code

File: app/components/machine/driver-custom/component.js [hunk 2]
```diff
@@ -16,12 +18,41 @@ export default Ember.Component.extend(ManageLabels, {
...
+        if(!labels){
+          labels={};
+        }
+        for(var tmp in labels){
+          out[tmp]=labels[tmp];
+        }
+        var subnet=this.get('subnet');
+        var routerIp=this.get('routerIp');
+        if (subnet) {
+          out['io.rancher.network.per_host_subnet.subnet']=subnet;
```
Comment (vincent99) [inline app/components/machine/driver-custom/component.js]: Put into constants (and the other below)

File: app/components/machine/driver-custom/component.js [hunk 3]
```diff
@@ -16,12 +18,41 @@ export default Ember.Component.extend(ManageLabels, {
     setLabels(labels) {
       if ( this.get('model') ) {
         var out = {};
+        var oldlabels=this.get('model.labels');
```
Comment (vincent99) [inline app/components/machine/driver-custom/component.js]: `const out = Object.assign({}, this.get('model.labels'));` (and below)

File: app/components/machine/driver-custom/component.js (hunk 1 above)
Comment (deniseschannon) [reply app/components/machine/driver-custom/component.js]: @orangedeng let's change this to `windowsCommand`

**Pull request merged**